### PR TITLE
Add source for video player and provision with terraform

### PR DESCRIPTION
**Pull request by benhylau**
Tested ipfs-mirror-0 on Android, it works. Please see inline comments.

### CODE REVIEW

File: terraform/ipfs-mirror/bootstrap.sh [hunk 1]
```diff
@@ -52,17 +60,16 @@ systemctl daemon-reload
...
+# Install web video player
+rm -rf /var/www/html/*
+cp -r /tmp/video-player/* /var/www/html/
 
-echo "originalgw='http://ipfs-server.$DOMAIN_NAME:8080/';" >> /var/www/html/common.js
-echo "gw='http://ipfs-mirror.$DOMAIN_NAME:8080/';" >> /var/www/html/common.js
-echo "ipnsm3u8='http://ipfs-server.$DOMAIN_NAME:8080/ipns/$IPFS_ID';" >> /var/www/html/common.js
-echo "clearm3u8='';" >> /var/www/html/common.js
+# Configure video player
+sed -i "s#__IPFS_GATEWAY_SELF__#http://ipfs-server.${DOMAIN_NAME}:8080#g" /var/www/html/common.js
```
Comment (benhylau) [inline terraform/ipfs-mirror/bootstrap.sh]: It's more clear to replace a template

File: terraform/shared/video-player/common.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// JavaScript Document
```
Comment (benhylau) [inline terraform/shared/video-player/common.js]: @darkdrgn2k please review

File: terraform/main.tf [hunk 3]
```diff
@@ -200,9 +208,9 @@ resource "digitalocean_record" "ipfs-mirror-v6" {
 output "digital_ocean_droplets" {
   depends_on = ["digitalocean_record.*"]
   value      = [
-    "${digitalocean_droplet.rtmp-server.name}: ${digitalocean_droplet.rtmp-server.status}",
-    "${digitalocean_droplet.ipfs-server.name}: ${digitalocean_droplet.ipfs-server.status}",
-    "${element(digitalocean_droplet.ipfs-mirror.*.name, 1)}: ${length(digitalocean_droplet.ipfs-mirror.*.status)} instance(s)",
+    "${digitalocean_droplet.rtmp-server.name}:             ${digitalocean_droplet.rtmp-server.status}",
+    "${digitalocean_droplet.ipfs-server.name}:             ${digitalocean_droplet.ipfs-server.status}",
+    "ipfs-mirror instance(s): ${length(digitalocean_droplet.ipfs-mirror.*.status)}",
```
Comment (benhylau) [inline terraform/main.tf]: @darkdrgn2k this should fix the destroy issue

File: terraform/shared/video-player/index.html [hunk 4]
```diff
@@ -0,0 +1,110 @@
+<!DOCTYPE html>
```
Comment (benhylau) [inline terraform/shared/video-player/index.html]: @darkdrgn2k made changes to this, please review
@Shrinks99 this and the css are the files you want to make pretty

File: terraform/shared/video-player/common.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// JavaScript Document
+ipfs_gateway_self='__IPFS_GATEWAY_SELF__';  // IPFS gateway of this node
+ipfs_gateway_origin='__IPFS_GATEWAY_ORIGIN__';  // IPFS gateway of origin stream
+m3u8_ipfs='__IPFS_GATEWAY_ORIGIN__/ipns/__IPFS_ID_ORIGIN__';  // URL to m3u8 over IPFS gateway
+m3u8_http='__HTTP_M3U8__';  // Optional URL to m3u8 over HTTP
```
Comment (darkdrgn2k) [inline terraform/shared/video-player/common.js]: Needs to be an array, this is how @ASoTNetworks  does "CDN" by randomly picking one of the gateways to stream from.

Comment (darkdrgn2k) [reply terraform/shared/video-player/common.js]: `m3u8_http=['server1','sever2','server3'];`

File: terraform/shared/video-player/index.html [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      videojs.options.html5.nativeAudioTracks=false;
+      videojs.options.html5.nativeVideoTracks=false;	
+      videojs.options.hls.overrideNative=true;
+
+      var live = videojs('live');
+      live.dvrseekbar();
+
+      function httpStream() {
+        live.src({
+          src: m3u8_http[Math.floor(Math.random() * m3u8_http.length)],
```
Comment (darkdrgn2k) [inline terraform/shared/video-player/index.html]: This is the random server picker i mentioned above.

**Pull request merged**